build.rs

```diff
@@ -0,0 +1,56 @@
+//! Reads the `templates/` directory and includes all versions' templates as
+//! part of the binary to reduce installation footprint
+extern crate ron;
+
+use std::env;
+use std::fs::{read_dir, File};
+use std::io::Write;
+use std::path::{Path, PathBuf};
+
+use ron::de::from_reader;
+
+fn path(env: &str, s: &str) -> PathBuf {
+    PathBuf::from(env::var(env).unwrap()).join(s)
+}
+
+fn read_template_index<P: AsRef<Path>>(p: P) -> Vec<String> {
```

Comment:
This should probably take `P: Into<PathBuf>` and use `p.into()` below to create the `PathBuf`.